### ISSUE TITLE
UV Editor - Edit Mode - Dynamic Select Similar menu needs to be in Header

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#3204]
![image](https://user-images.githubusercontent.com/35357409/185995133-2f07a183-1df0-493d-9ac7-ebe79f1f5a49.png)

The Select Similar operator in the header of the UV Editor needs to be dynamic to show the operators. 

This is a continuation of #3088 - maybe another hand can make this aspect work.

**Comment by ReinerBforartists** [comment]
Draise, i had already a crack at it. Maybe Iyad knows a solution.

**Comment by Draise14** [comment]
Gonna wait on merge, then try tackle this.. I had an enumerated if/else in the sequencer that might come in handy as to show how.

**Comment by Draise14** [comment]
![bforartists_KtaWo0d73O](https://user-images.githubusercontent.com/35357409/227073183-111690fd-6f7b-4fc2-be0f-9d04cb44c34d.gif)

OK! Getting somewhere.

**Comment by Draise14** [comment]
I'm really close.. but I got an issue...

I'm using a "try" python effort with a conditional to call the right operators - but there is a tuple or list of enums on the uv.select_simular operator that don't load till I use the floating menu. This causes an exception. My workaround is throw in a warning, because after using the floating panel.. it's fine? 

This causes an incomplete operation list on first use... 
The issue is: `TypeError: bpy_struct: item.attr = val: enum "LENGTH" not found in ('PIN')`
Hinting that the enumerated list for the other type methods are not populated on first use, only on the default. 

![bforartists_SH2fNfQ9in](https://user-images.githubusercontent.com/35357409/227102232-aa5ddcfa-02d7-4676-a58d-868f838f7ece.gif)

# New Code:
This is what I got, a new menu. 
```### BFA - Changes start - added a new menu for the select similar in the header GUI. To avoid issues, the floating menu operator is included. 
class IMAGE_MT_select_similar(Menu):
    bl_label = "Select Similar"

    def draw(self, context):
        layout = self.layout

        mode = bpy.context.tool_settings.uv_select_mode
       
        #data_selection = ['VERTEX', 'EDGE', 'FACE', 'ISLAND']
        #type_selection = ['PIN', 'LENGTH', 'LENGTH_3D', 'AREA', 'AREA_3D', 'POLYGON_SIDES', 'MATERIAL']               
        
        try:
            layout.operator_context = 'INVOKE_REGION_WIN'
            
            if mode == 'VERTEX':
                #Detects if it's in Vertex mode, the default
                layout.operator_context = 'INVOKE_DEFAULT'
                layout.operator("uv.select_similar", text="Floating Menu", icon="SIMILAR")
                layout.separator()
                layout.operator("uv.select_similar", text="Pinned", icon="PINNED").type = "PIN"
            elif mode == 'EDGE':
                #Detects if it's in Edge mode
                layout.operator_context = 'INVOKE_DEFAULT'
                layout.operator("uv.select_similar", text="Floating Menu", icon="SIMILAR")
                layout.separator()
                layout.operator("uv.select_similar", text="Length", icon="PARTICLEBRUSH_LENGTH").type = "LENGTH"
                layout.operator("uv.select_similar", text="Length 3D", icon="PARTICLEBRUSH_LENGTH").type = "LENGTH_3D"
                layout.operator("uv.select_similar", text="Pinned", icon="PINNED").type = "PIN"
            elif mode == 'FACE':
                #Detects if it's in Face mode
                layout.operator_context = 'INVOKE_DEFAULT'
                layout.operator("uv.select_similar", text="Floating Menu", icon="SIMILAR")
                layout.separator()
                layout.operator("uv.select_similar", text="Area", icon="AREA").type = "AREA"
                layout.operator("uv.select_similar", text="Area 3D", icon="AREA").type = "AREA_3D"
                layout.operator("uv.select_similar", text="Material", icon="MATERIAL_DATA").type = "MATERIAL"
                layout.operator("uv.select_similar", text="Object", icon="OBJECT_DATA").type = "OBJECT"
                layout.operator("uv.select_similar", text="Polygon Sides", icon="SELECT_FACES_BY_SIDE").type = "SIDES"
                layout.operator("uv.select_similar", text="Winding", icon="WINDING").type = "WINDING"
            else:
                #Detects if it's in Island mode
                layout.operator("uv.select_similar", text="Floating Menu", icon="SIMILAR")
                layout.separator()
                layout.label(text="Please use another selection method")
        except:
            print("A uv.select_similar exception occurred")
            # Right now an exception happens before you activate the enum list with the floating panel. Till the activation you'll get a warning hint.
            # This causes an incomplete operation list on first use... 
            # The issue is: "TypeError: bpy_struct: item.attr = val: enum "LENGTH" not found in ('PIN')" 
            # Hinting that the enumerated list for the other type methods are not populated on first use, only on the default Vertex enums ('PIN').
            layout.label(text="Use the operator to activate all options")
        
### BFA - End of changes
```

This works on a new menu entry in the `class IMAGE_MT_select(Menu):`:

```
        layout.menu("IMAGE_MT_select_similar") #BFA - new menu to add to header
```

I feel I am close, I just don't know how to make it list all the enums per selection type on first use.... Something might need to be registered better? Not sure how the floating menu populates. Maybe I need to figure that out... any hints?

**Comment by Draise14** [comment]
I pushed this to see... Maybe it's good enough to pull? 

![image](https://user-images.githubusercontent.com/35357409/227104631-d9d8db86-1f0c-4f68-9fd7-b68b32e88232.png)

**Update:**
- Added in a dynamic menu, it works perfectly after first use. 
- It won't populate till you use the floating menu once, but I added a hint
- May need work, but for now it's somewhat an improvement
- Added a GUI entry to the floating panel nonetheless.

**Comment by ReinerBforartists** [comment]
I'm lost with this, sorry :)

**Comment by Draise14** [comment]
Should we leave as is for now?

**Comment by ReinerBforartists** [comment]
Your decision

**Comment by Draise14** [comment]
I have more clues to experiment with this one, so won't close yet.

**Comment by ReinerBforartists** [comment]
Just keep an eye at the rest of the tracker too. There is more important things to fix than this one.

**Comment by Draise14** [comment]
Ok, yeah, bad timing on my end..  I will commit. it only really is an issue on first use, then it is as expected. 

I'm trying to stick to one task at a time - I have a lot on my plate.

**Comment by ReinerBforartists** [comment]
Draise could you please revert this as long as you haven't solved all problems with it? This is crucial. When it does not work at first use then it does not work at all.

**Comment by ReinerBforartists** [comment]
I have reverted the change now. Please first make it 100% working.

**Comment by Draise14** [comment]
Ah ok. I did ask if we should leave as is. You seemed ok with it. It did have a warning. 

I'll keep at it next week.

**Comment by ReinerBforartists** [comment]
As told, sorry that this slipped though. I cannot have my eyes everywhere. And i have overread the part with the first use. This is crucial. Because the user expect it to work at first use.

**Comment by Draise14** [comment]
Postponed for 3.6.0
